### ISSUE TITLE
fix issue mhwd not configuring properly with recent manjaro-tools changes

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/release-plan#33]
With current build **kde-minimal 20160228** I found a regression with **sddm**. It works with autologin but never starts on user login. We have followed error:

```
● sddm.service - Simple Desktop Display Manager
   Loaded: loaded (/usr/lib/systemd/system/sddm.service; enabled; vendor preset: disabled)
   Active: inactive (dead) (Result: core-dump) since Sun 2016-02-28 20:41:17 AWST; 3min 9s ago
     Docs: man:sddm(1)
           man:sddm.conf(5)
  Process: 710 ExecStart=/usr/bin/sddm (code=dumped, signal=SEGV)
 Main PID: 710 (code=dumped, signal=SEGV)

Feb 28 20:41:17 pc systemd[1]: sddm.service: Main process exited, code=dumped, status=11/SEGV
Feb 28 20:41:17 pc systemd[1]: sddm.service: Unit entered failed state.
Feb 28 20:41:17 pc systemd[1]: sddm.service: Failed with result 'core-dump'.
Feb 28 20:41:17 pc systemd[1]: sddm.service: Service hold-off time over, scheduling restart.
Feb 28 20:41:17 pc systemd[1]: Stopped Simple Desktop Display Manager.
Feb 28 20:41:17 pc systemd[1]: sddm.service: Start request repeated too quickly.
Feb 28 20:41:17 pc systemd[1]: Failed to start Simple Desktop Display Manager.
```
Build **kde-minimal 20160220** however was fine. Not so few updates during the week. Most likely it is **glibc**.

**Comment by philmmanjaro** [comment]
We have followed package diff:

```
--- kde-20160220.pkgs.txt	2016-02-28 13:51:32.616755894 +0100
+++ kde-20160228.pkgs.txt	2016-02-28 13:47:36.105240508 +0100
@@ -3,9 +3,9 @@ acpid 2.0.26-1
 adwaita-icon-theme 3.18.0-1
 alsa-firmware 1.0.29-1
 alsa-lib 1.1.0-1
-alsa-plugins 1.1.0-1
+alsa-plugins 1.1.0-2
 alsa-utils 1.1.0-1
-archlinux-keyring 20160123-1
+archlinux-keyring 20160215-1
 aspell 0.60.6.1-4
 at-spi2-atk 2.18.1-1
 at-spi2-core 2.18.3-1
@@ -28,11 +28,11 @@ btrfs-progs 4.4-1
 bzip2 1.0.6-5
 ca-certificates 20150402-1
 ca-certificates-cacert 20140824-2
-ca-certificates-mozilla 3.22-1
+ca-certificates-mozilla 3.22.1-1
 ca-certificates-utils 20150402-1
 cairo 1.14.6-1
-calamares-dev 1.98.0.r2022.12e20f6-1
-cantarell-fonts 1:0.0.22-1
+calamares 2.0-1
+cantarell-fonts 1:0.0.23-1
 cdparanoia 10.2-5
 cifs-utils 6.4-1
 ckbcomp 1.120-1
@@ -45,7 +45,7 @@ cpupower 4.4-1
 cracklib 2.9.4-1
 crda 3.18-1
 cronie 1.5.0-1
-cryptsetup 1.7.0-1
+cryptsetup 1.7.0-2
 curl 7.47.1-1
 damageproto 1.2.1-3
 db 5.3.28-3
@@ -83,11 +83,11 @@ exfat-utils 1.2.3-2
 exiv2 0.25-3
 expat 2.1.0-4
 f2fs-tools 1.6.0-1
-ffmpeg 1:2.8.6-2
+ffmpeg 1:3.0-1
 fftw 3.3.4-1
 file 5.25-1
 filesystem 2015.09-1
-findutils 4.4.2-6
+findutils 4.6.0-1
 fixesproto 5.0-3
 flac 1.3.1-3
 fontconfig 2.11.1-2
@@ -100,18 +100,18 @@ fuse 2.9.5-1
 gamin 0.1.10-8
 gawk 4.1.3-1
 gc 7.4.2-4
-gcc-libs-multilib 5.3.0-4
-gdb 7.10.1-1
+gcc-libs-multilib 5.3.0-5
+gdb 7.11-1
 gdbm 1.11-1
 gdk-pixbuf2 2.32.3-1
 geoclue 0.12.99-2
 geoclue2 2.4.1-1
-gettext 0.19.6-2
+gettext 0.19.7-1
 giflib 5.1.2-1
 glew 1.13.0-1
 glib-networking 2.46.1-1
-glib2 2.46.2-2
-glibc 2.22-4
+glib2 2.46.2-4
+glibc 2.23-1
 glu 9.0.0-4
 gmp 6.1.0-3
 gnome-icon-theme 3.12.0-3
@@ -135,13 +135,13 @@ gst-plugins-base 1.6.3-1
 gst-plugins-base-libs 1.6.3-1
 gstreamer 1.6.3-1
 gtk-theme-maia 5.5.2-1
-gtk-update-icon-cache 3.18.7-2
+gtk-update-icon-cache 3.18.8-1
 gtk2 2.24.29-1
-gtk3 3.18.7-2
+gtk3 3.18.8-1
 guile 2.0.11-3
 gzip 1.6-1
-harfbuzz 1.2.0-1
-harfbuzz-icu 1.2.0-1
+harfbuzz 1.2.3-1
+harfbuzz-icu 1.2.3-1
 haveged 1.9.1-2
 hdparm 9.48-1
 hicolor-icon-theme 0.15-1
@@ -155,10 +155,10 @@ ilmbase 2.2.0-2
 inetutils 1.9.4-2
 inputproto 2.3.1-1
 intel-ucode 20151106-1
-inxi 2.2.33-1
+inxi 2.2.34-1
 iproute2 4.1.1-1
 iptables 1.4.21-3
-iputils 20140519.fad11dc-1
+iputils 20150815.1c59920-3
 ipw2100-fw 1.3-7
 ipw2200-fw 3.1-5
 iso-codes 3.65-1
@@ -199,7 +199,7 @@ kdnssd 5.19.0-1
 kdoctools 5.19.0-1
 kemoticons 5.19.0-1
 keyutils 1.5.9-1
-kfilemetadata 5.19.0-1
+kfilemetadata 5.19.0-2
 kglobalaccel 5.19.0-1
 kguiaddons 5.19.0-1
 khelpcenter 5.5.4-1
@@ -227,7 +227,7 @@ kpackage 5.19.0-1
 kparts 5.19.0-1
 kpeople 5.19.0-1
 kplotting 5.19.0-1
-kpmcore 2.0.0-1
+kpmcore 2.0.1-1
 kpty 5.19.0-1
 krb5 1.13.2-1
 krunner 5.19.0-1
@@ -261,9 +261,9 @@ lib32-attr 2.4.47-1
 lib32-bzip2 1.0.6-2
 lib32-elfutils 0.165-1
 lib32-expat 2.1.0-2
-lib32-gcc-libs 5.3.0-4
+lib32-gcc-libs 5.3.0-5
 lib32-glew 1.13.0-1
-lib32-glibc 2.22-4
+lib32-glibc 2.23-1
 lib32-glu 9.0.0-3
 lib32-libcap 2.25-1
 lib32-libdrm 2.4.67-1
@@ -293,14 +293,14 @@ lib32-mesa 11.1.2-1
 lib32-mesa-demos 8.2.0-2
 lib32-ncurses 6.0-2
 lib32-readline 6.3.008-2
-lib32-systemd 228-1
+lib32-systemd 229-0.1
 lib32-util-linux 2.27.1-1
 lib32-wayland 1.10.0-1
 lib32-xz 5.2.2-1
 lib32-zlib 1.2.8-1
 libaio 0.3.110-1
 libarchive 3.1.2-8
-libass 0.13.1-1
+libass 0.13.2-1
 libassuan 2.4.2-1
 libasyncns 0.8-5
 libatasmart 0.19-3
@@ -359,8 +359,8 @@ libpgm 5.2.122-2
 libpipeline 1.4.1-1
 libplist 1.12-4
 libpng 1.6.21-1
-libproxy 0.4.11-8
-libpulse 8.0-1
+libproxy 0.4.12-1
+libpulse 8.0-3
 libqalculate 0.9.7-5
 libqmi 1.12.8-1
 libraw1394 2.1.1-1
@@ -375,9 +375,9 @@ libsndfile 1.0.26-1
 libsodium 1.0.8-1
 libsoup 2.52.2-1
 libsoxr 0.1.2-1
-libssh 0.7.2-1
-libssh2 1.6.0-1
-libsystemd 229-2
+libssh 0.7.3-1
+libssh2 1.7.0-1
+libsystemd 229-3
 libtasn1 4.7-1
 libteam 1.23-1
 libthai 0.1.22-1
@@ -399,10 +399,10 @@ libvisual 0.4.0-6
 libvoikko 4.0.2-1
 libvorbis 1.3.5-1
 libvpx 1.5.0-4
-libwbclient 4.3.4-2
+libwbclient 4.3.5-1
 libwebp 0.5.0-1
 libx11 1.6.3-1
-libx264 2:148.20150725-1
+libx264 2:148.20160103-1
 libx86emu 1.1.23.1-1
 libxau 1.0.8-2
 libxaw 1.0.13-1
@@ -437,11 +437,11 @@ libxxf86vm 1.1.4-1
 libyaml 0.1.6-1
 libzip 1.0.1-1
 licenses 20140629-1
-linux-api-headers 4.1.4-1
+linux-api-headers 4.4.1-1
 linux-firmware 20160113.40e9ae8-1
-linux44 4.4.2-1
-linux44-ndiswrapper 1.59-3
-linux44-virtualbox-guest-modules 5.0.14-3
+linux44 4.4.3-1
+linux44-ndiswrapper 1.59-4
+linux44-virtualbox-guest-modules 5.0.14-4
 llvm-libs 3.7.1-1
 lm_sensors 3.4.0-1
 lmdb 0.9.18-1
@@ -452,7 +452,7 @@ lvm2 2.02.141-1
 lz4 131-1
 lzo 2.09-1
 lzop 1.03-5
-maia-icon-theme 0.5.6-1
+maia-icon-theme 0.5.6-2
 man-db 2.7.5-1
 man-pages 4.04-1
 manjaro-alsa 2012.11-1
@@ -462,19 +462,20 @@ manjaro-hotfixes 2015.12-2
 manjaro-kde-minimal-settings 20160216-1
 manjaro-keyring 20150809-1
 manjaro-live-skel 20160202-1
-manjaro-livecd-cli-installer 20151210-1
-manjaro-livecd-systemd 20151210-1
+manjaro-livecd-base 20160222-3
+manjaro-livecd-cli-installer 20160225-1
+manjaro-livecd-systemd 20160222-1
 manjaro-release-dev 16.06-0
 manjaro-system 20151213-2
-manjaro-tools-base 0.10.3-1
-manjaro-welcome 15.12-2
-mdadm 3.3.4-1
+manjaro-tools-base 0.11.0-2
+manjaro-welcome 15.12-3
+mdadm 3.4-1
 media-player-info 22-1
 memtest86+ 5.01-1
 mesa 11.1.2-1
 mesa-demos 8.3.0-1
 mhwd 0.5.6-7
-mhwd-amdgpu 1.0.0-1
+mhwd-amdgpu 1.0.1-1
 mhwd-catalyst 1:15.201.1151-1
 mhwd-db 0.5.6-12
 mhwd-nvidia 1:352.79-1
@@ -503,14 +504,14 @@ nilfs-utils 2.2.3-1
 noto-fonts 20151224-2
 noto-fonts-emoji 20151224-1
 npth 1.2-1
-nspr 4.11-1
-nss 3.22-1
+nspr 4.12-1
+nss 3.22.1-1
 nss-mdns 0.10-6
 ntfs-3g 2015.3.14-1
 ntp 4.2.8.p6-3
 numlockx 1.2-3
 octopi 0.8.0-4
-openal 1.17.2-1
+openal 1.17.2-2
 opencore-amr 0.1.3-2
 openexr 2.2.0-3
 openjpeg 1.5.2-1
@@ -521,14 +522,14 @@ opus 1.1.2-1
 orc 0.4.24-1
 os-prober 1.70-1
 p11-kit 0.23.2-1
-pacman 5.0.0-1
-pacman-mirrorlist 20160208-1
+pacman 5.0.1-2
+pacman-mirrorlist 20160227-1
 pam 1.2.1-3
 pam_encfs 0.1.4.4-2
 pambase 20130928-1
 pango 1.38.1-1
 parted 3.2-3
-partitionmanager 2.0.0-1
+partitionmanager 2.0.1-1
 pciutils 3.4.0-1
 pcmciautils 018-7
 pcre 8.38-2
@@ -539,7 +540,7 @@ pinentry 0.9.7-1
 pixman 0.34.0-1
 pkg-config 0.29-1
 plasma-desktop 5.5.4-1.1
-plasma-framework 5.19.0-2
+plasma-framework 5.19.0-3
 plasma-nm 5.5.4-1
 plasma-pa 5.5.4-1
 plasma-workspace 5.5.4-1
@@ -560,7 +561,7 @@ prison-frameworks 1.2.1.20150831-2
 procps-ng 3.3.11-2
 progsreiserfs 0.3.0.5-8
 psmisc 22.21-3
-pulseaudio 8.0-1
+pulseaudio 8.0-3
 pv 1.6.0-2
 pygobject-devel 3.18.2-1
 pygobject2-devel 2.28.6-12
@@ -611,7 +612,7 @@ rp-pppoe 3.12-1
 rpcbind 0.2.3-3
 rsync 3.1.2-1
 rtkit 0.11-5
-run-parts 4.5.1-1
+run-parts 4.7-1
 s-nail 14.8.6-1
 sbc 1.3-1
 schroedinger 1.0.11-3
@@ -626,19 +627,19 @@ shadow 4.2.1-3
 shared-mime-info 1.5-1
 slang 2.3.0-1
 smartmontools 6.4-2
-smbclient 4.3.4-2
+smbclient 4.3.5-1
 solid 5.19.0-1
 sonnet 5.19.0-1
 sound-theme-freedesktop 0.8-2
 speex 1.2rc2-1
 speexdsp 1.2rc3-2
-sqlite 3.10.2-1
+sqlite 3.11.0-2
 squashfs-tools 4.3-4
 sudo 1.8.15-1
 sysfsutils 2.1.0-9
-syslinux 6.03-4.2
-systemd 229-2
-systemd-sysvcompat 229-2
+syslinux 6.03-5
+systemd 229-3
+systemd-sysvcompat 229-3
 systemsettings 5.5.4-1
 taglib 1.10-3
 talloc 2.1.5-1
@@ -751,7 +752,7 @@ xorg-xlsclients 1.1.3-1
 xorg-xmessage 1.0.4-2
 xorg-xmodmap 1.0.9-1
 xorg-xprop 1.2.2-1
-xorg-xrandr 1.4.3-1
+xorg-xrandr 1.5.0-1
 xorg-xrdb 1.1.0-2
 xorg-xrefresh 1.0.5-1
 xorg-xset 1.2.3-1
```

**Comment by philmmanjaro** [comment]
Seems to be something else. Here is [more information](https://gist.github.com/philmmanjaro/2d3ce60c16098833bd0e) about it.

**Comment by philmmanjaro** [comment]
More or less this is the issue:
```
Feb 28 13:52:00 john-pc sddm[654]: Adding new display on vt 1 ...
Feb 28 13:52:00 john-pc sddm[654]: Display server starting...
Feb 28 13:52:00 john-pc sddm[654]: Running: /usr/bin/X -nolisten tcp -auth /var/run/sddm/{cb1bdd3c-67a6-498a-b633-039d1d9442a3} -background none -noreset -displayfd 18 vt1
Feb 28 13:52:00 john-pc sddm-helper[672]: [PAM] Starting...
Feb 28 13:52:00 john-pc sddm-helper[672]: [PAM] Authenticating...
Feb 28 13:52:00 john-pc sddm-helper[672]: [PAM] returning.
Feb 28 13:52:00 john-pc systemd-coredump[664]: Process 591 (sddm-greeter) of user 995 dumped core.
#12 0x0000000000429c60 _ZN4SDDM10GreeterAppC1ERiPPc (sddm-greeter)
#13 0x00000000004143d6 main (sddm-greeter)
#15 0x00000000004144b9 _start (sddm-greeter)
Feb 28 13:52:00 john-pc kernel: sddm[654]: segfault at 0 ip 00007fb1de070ef3 sp 00007ffc809b7070 error 4 in libQt5Core.so.5.5.1[7fb1dddb5000+4c3000]
Feb 28 13:52:00 john-pc sddm[654]: Running display setup script "/usr/share/sddm/scripts/Xsetup"
Feb 28 13:52:00 john-pc sddm[654]: /usr/bin/xauth: (stdin):1: bad "remove" command line
Feb 28 13:52:00 john-pc sddm[654]: /usr/bin/xauth: (stdin):2: bad "add" command line
Feb 28 13:52:00 john-pc sddm[654]: Display server started.
Feb 28 13:52:00 john-pc sddm[654]: Socket server starting...
Feb 28 13:52:00 john-pc sddm[654]: Socket server started.
Feb 28 13:52:00 john-pc sddm[654]: Greeter starting...
Feb 28 13:52:00 john-pc sddm[654]: Adding cookie to "/var/run/sddm/{cb1bdd3c-67a6-498a-b633-039d1d9442a3}"
Feb 28 13:52:00 john-pc sddm-helper[680]: [PAM] Starting...
Feb 28 13:52:00 john-pc sddm-helper[680]: [PAM] Authenticating...
Feb 28 13:52:00 john-pc sddm-helper[680]: [PAM] returning.
Feb 28 13:52:00 john-pc sddm[654]: QProcess: Destroyed while process ("/usr/lib/sddm/sddm-helper") is still running.
Feb 28 13:52:00 john-pc systemd[1]: sddm.service: Main process exited, code=dumped, status=11/SEGV
Feb 28 13:52:00 john-pc sddm-helper[680]: Received a wrong opcode instead of AUTHENTICATED: 0
Feb 28 13:52:00 john-pc systemd[1]: sddm.service: Unit entered failed state.
Feb 28 13:52:00 john-pc systemd[1]: sddm.service: Failed with result 'core-dump'.
Feb 28 13:52:00 john-pc systemd[1]: sddm.service: Service hold-off time over, scheduling restart.
Feb 28 13:52:00 john-pc systemd[1]: sddm.service: Start request repeated too quickly.
Feb 28 13:52:00 john-pc systemd-coredump[683]: Process 654 (sddm) of user 0 dumped core.
#1 0x0000000000430895 _ZN4SDDM4Auth7Private9setSocketEP12QLocalSocket (sddm)
#2 0x00000000004325e4 _ZN4SDDM4Auth12SocketServer19handleNewConnectionEv (sddm)
#16 0x000000000041b146 main (sddm)
#18 0x000000000041b429 _start (sddm)
```

Graphics driver related?

**Comment by philmmanjaro** [comment]
Related to https://github.com/sddm/sddm/issues/556

**Comment by Kirek** [comment]
Could you reproduce this on a installed system, or is virtualbox only issue?

**Comment by philmmanjaro** [comment]
I'm just testing this with virtualbox. I can try with vmware later. However I've no PC or time to install it on a real system. I'm trying to do now a stable branch ISO and check if some of the new packages introduces this ...

**Comment by Kirek** [comment]
Tried if the last update breaks my kde installations: laptop and vmware are fine.
Building a iso now to test live...

**Comment by udeved** [comment]
I can report zero issue with sddm and nvdia and bumblebee openrc systems.
But, its sddm-consolekit package.

**Comment by udeved** [comment]
@Kirek 

Can you say if the kde power management override settings in the live overlay are necessary?

**Comment by Kirek** [comment]
They are necessary, so it doesn't lock the screen will installing.

**Comment by udeved** [comment]
Ah yes, the default values to screen lock are ridiculously short, I noticed om fresh laptop install.

**Comment by philmmanjaro** [comment]
Seems to be somehow a regression with **manjaro-tools** I was kinda able to build a working ISO with **manjaro-tools-iso-git r1893.3c77b38-1**. So something in regard of display manager setup has changed. @udeved: can you review all your changes you have made in that regard?

**Comment by udeved** [comment]
Will do, but I actually changed nothing in the systemd behavior, just reorganized plymouth, and added some non-systemd tweaks for sddm.

Gonna build kde-minimal-systemd for test.

**Comment by udeved** [comment]
~~~
 --> Configuring [systemd]
  -> Setting cronie ...
  -> Setting ModemManager ...
  -> Setting NetworkManager ...
  -> Setting tlp ...
  -> Setting tlp-sleep ...
  -> Setting sddm ...
 --> Done configuring [systemd]
~~~

Looks good so far, its without plymouth.

**Comment by udeved** [comment]
Ok, first test, lxqt-minimal-systemd passed.

sddm starts and autologin works.

Will repeat with kde-minimal

**Comment by udeved** [comment]
Second test, kde-minimal as is from git.

~~~
 --> Configuring [systemd]
  -> Setting avahi-daemon ...
  -> Setting bluetooth ...
  -> Setting cronie ...
  -> Setting ModemManager ...
  -> Setting NetworkManager ...
  -> Setting tlp ...
  -> Setting tlp-sleep ...
  -> Setting sddm ...
 --> Done configuring [systemd]
~~~

**Comment by udeved** [comment]
Second test passed too.

Last test, kde-minimal with autologin and plymouth disabled

Btw, sddm hangs terribly on shutdow with systemd.

**Comment by udeved** [comment]
3rd test passed.

![sddm] (http://s29.postimg.org/rlz14oczb/sddm.png)

**Comment by philmmanjaro** [comment]
Please install your image in virtualbox. Live-session works. However not if you install it without auto-login. That needs tested ...

**Comment by udeved** [comment]
Installing...

A shot into the blue, its nothing to do with cal user=root pwd setting?

**Comment by philmmanjaro** [comment]
Somehow **/x86_64/kde-minimal-image/etc/systemd/logind.conf** is missing in the latest ISOs.

**Comment by udeved** [comment]
~~~
configure_systemd(){
	if [[ ${initsys} == 'systemd' ]];then
		msg2 "Configuring logind"
		local conf=$1/etc/systemd/logind.conf
		sed -i 's/#\(HandleSuspendKey=\)suspend/\1ignore/' "$conf"
		sed -i 's/#\(HandleLidSwitch=\)suspend/\1ignore/' "$conf"
	fi
}
~~~

Its properly called for root-image.

If cal will gain root-image only install, you'd want it logind stuff on root-image, right? ;)
I moved it to root-image, and live-image manipulates it further.

**Comment by philmmanjaro** [comment]
Ok, found it. Seems ok. However somehow it stuck on installed systems ...

**Comment by udeved** [comment]
I can confirm. its stuck at plymouth, despite plymouth beinng diabled on my test iso.

**Comment by udeved** [comment]
Perhaps its due to logind.conf on root-image, which is lowest mount layer?

**Comment by udeved** [comment]
For me, its stuck after initramfs is run and systemd is supposed to do its thing.

**Comment by udeved** [comment]
The unit file looks wrong, if sddm and not sddm-plymouth is set.

~~~
[Unit]
Description=Simple Desktop Display Manager
Documentation=man:sddm(1) man:sddm.conf(5)
Conflicts=getty@tty1.service
After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service

[Service]
ExecStart=/usr/bin/sddm
Restart=always
#PrivateTmp=yes

[Install]
Alias=display-manager.service
~~~

**Comment by philmmanjaro** [comment]
There is no **sddm-plymouth** ...

**Comment by udeved** [comment]
No binary, but a unit file.

**Comment by udeved** [comment]
I mean, why is plymouth.quit called?
Does plymouth start on systemd regardless of any settings?

**Comment by philmmanjaro** [comment]
I ment there is no **sddm-plymouth.service** file

**Comment by udeved** [comment]
But there should be for plymouth, right?

**Comment by philmmanjaro** [comment]
Well, **manjaro-tools-iso-git r1893.3c77b38-1** works just fine. However not later versions of **manjaro-tools**. I'll try to test some more ...

**Comment by udeved** [comment]
![plymouth] (http://s15.postimg.org/pinn6mn8r/plymouth.png)

Will check plymouth settings again.

But what logic is behind trying to quite plymouth, if it is deactivated?

**Comment by udeved** [comment]
~~~
if [[ ${displaymanager} != "none" ]];then
				local service=${displaymanager}
				if ${plymouth_boot}; then
					#msg2 "Setting plymouth %s ...." "${plymouth_theme}"
					sed -i -e "s/^.*Theme=.*/Theme=${plymouth_theme}/" $1/etc/plymouth/plymouthd.conf
					if [[ -f $1/etc/systemd/system/${displaymanager}-plymouth.service ]] || \
					[[ -f $1/usr/lib/systemd/system/${displaymanager}-plymouth.service ]];then
						service=${displaymanager}-plymouth
					fi
				fi
				add_svc_sd "$1" "$service"
			fi
~~~

Cab see any wrong here.

**Comment by philmmanjaro** [comment]
Well, it seems to be **manjaro-tools** and not some packages. Whatever changed after **r1893.3c77b38** "broke" **sddm** somehow. I've to check what I've to rollback in my profiles to test more.

**Comment by udeved** [comment]
I suspect something plymouth, I am on it.
Will keep you updated to possibly test a fix.

**Comment by oberon2007** [comment]
This is obviously not related to *sddm* however. I get the same with Deepin and LightDM. Live in virtualbox everything works fine. Then installed in virtualbox and boot hangs after plymouth. Then I have just a blinking cursor...

**Comment by philmmanjaro** [comment]
Hmm, interesting. I'm also with some theory ...

**Comment by philmmanjaro** [comment]
So, I can confirm that **r1931.01e4382-1** of manjaro-tools also has this blocking issue ...

**Comment by udeved** [comment]
I think I fixed it.

I thik what happens here, is that plymouth somehow can't start or stop, because it can find the theme.
I made it so that the theme is only set if plymouth_boot=true, this was reverted.

https://github.com/manjaro/manjaro-tools/commit/b6b8669fc536b07ff41aa0fe9b8caef2456b8ff4

**Comment by udeved** [comment]
https://github.com/manjaro/manjaro-tools/commit/2be07641f34f6e57fc6cd66ab41a23e1029ac880

**Comment by udeved** [comment]
I believe we should clean the pkglist from plymouth, if plymouth_boot true, ie add >plymouth tag to pkglist.

**Comment by oberon2007** [comment]
Exactly the same issue for me still with **r1941.c6fd218**
Deepin with Plymouth and LightDm.

**Comment by udeved** [comment]
There is an issue with sddm and systemd.
It core dumps if the system is purged from plymouth, and the plymouth nonsense removed from the sddm unit file, and /etc/sddm.conf removed.
It should start up this way, but as mentioned... core-dump.

Also, the cal-2.0 seems to cunningly add plymouth to both /etc/default/grub and /etc/mkinitcpio.conf.
This should be configurable in the concerning cal modules.

**Comment by udeved** [comment]
https://github.com/calamares/calamares-manjaro/pull/7

**Comment by udeved** [comment]
https://github.com/manjaro/thus/pull/42

**Comment by oberon2007** [comment]
Still not solved for me with https://github.com/manjaro/manjaro-tools/commit/e009a7fbe9ab16c1a24778c25b6d5d7cedcb68b4
In case that helps: The issue was already present five days ago with https://github.com/manjaro/manjaro-tools/commit/60553d50ae7dc5ca19c582141cf675cf08abee6a

**Comment by udeved** [comment]
Naturally, this is solved when installers have been rebuilt.

**Comment by oberon2007** [comment]
I see... Good to know! ;)

**Comment by oberon2007** [comment]
Shall we push a new version of **thus** to test? Or do we need to wait for something else?

**Comment by udeved** [comment]
It simply slipped through to update installers, after I changed a mwhd repo path.
Trivial reason, but big effect, installers didn't install video drivers at install.

**Comment by udeved** [comment]
I'd wait for Phil to merge calamares pull request, and build them all at once.
Never touched installers except for cli-installer.

**Comment by oberon2007** [comment]
:yum:
ok.
What will happen then in case people update the system in the live environment of a 15.12 ISO - will the install still work properly for those, too?

**Comment by udeved** [comment]
Kirek kindly merged calamares just now. :)

**Comment by udeved** [comment]
@oberon2007 

what do you mean?
When the installers get updated, technically, the manjaro-tools-0.11.0 repo version will start to produce
a working installed system again. 
I already split out the live session relevant part in a separate package couple of days ago.

**Comment by udeved** [comment]
@philmmanjaro @Kirek 

Can you please have a look here
https://github.com/manjaro/manjaro-tools-livecd/tree/master/livecd-base/usr

In my view, we could outsource thus and clamares scripting, and slap a manjaro-livecd-base as additional depend on the packages. 

If there were any such changes in he future, we would just have to update manjaro-livecd-base package, which is kid of manjaro-tools extension for live sesssion.

**Comment by oberon2007** [comment]
@udeved I am talking about ISOs that have already been created with for example **manjaro-tools-git r1893.3c77b38**, or older ones that people download from SF currently. When they decide to update the system in the live environment, the installers will be updated, too. Will it be a problem to install an old ISO with a new installer? Will the 15.12 ISOs have to be rebuilt for that reason? Or does the change of mhwd path only concern the installers in themselves?

**Comment by udeved** [comment]
Ah, they would also produce a working install if the installers in live session were updated before launching install.

**Comment by Kirek** [comment]
The installer will be broken if you update calamares in 15.12 live in the same way it broken with the lates manjaro-tools-git and unfiexed installers. With 15.12 install medias they will search in /opt/live for the mhwd repo, and in this isos it is in /opt/livecd
The easier way will be to revert the changes to the installer (livecd to live) and revert this for manjaro-tools, at least for the mhwd repo.

**Comment by udeved** [comment]
Well, yes, 15.12 won't work of course.
Any iso created after the /opt/live change will work with updated packages.

**Comment by udeved** [comment]
I was asking, if the 3 installers shoudl have a common scripting base.
If you look at the scripts, they are similar.

**Comment by udeved** [comment]
I would be more in favour of releasing a 16.03.
It think we can do it while Phil has fun on the beaches.

**Comment by Ste74** [comment]
But why people have to update system before install? I have try only one
time and the space in usb drive is less and update abort....
Il 29/feb/2016 15:04, "udeved" <notifications@github.com> ha scritto:

**Comment by udeved** [comment]
@Ste74 

15.12 does not need to be updated before install.

It only effects the iso which are built with manjaro-tools-git after 15.12

**Comment by Ste74** [comment]
Ah ok.... Lol
Il 29/feb/2016 15:15, "udeved" <notifications@github.com> ha scritto:

**Comment by oberon2007** [comment]
People don't **have** to update, no. But they might, and some people do! And in that case they will not be able to install... that was my question.
So. I think in that case we should definitely produce updated install media as soon as we have updated installers in stable branch ...

**Comment by udeved** [comment]
Yes, we also need to update media to use settings packages, so users won't have to manually clea /etc/skel any longer.

**Comment by oberon2007** [comment]
It wouldn't necessarily need to be a huge release with annoucements and stuff - if we don't want to . Just the files need to be updated at least.
Release annoucements on the other hand are always good  for publicity :stuck_out_tongue_winking_eye: And we could present to the public how also desktop settings are now rolling release on Manjaro.
To provide [minimal editions](https://github.com/manjaro/release-plan/issues/34#issue-137264686) additionally is a great idea I think.
The resonance on cinnamon-minimal was quite positive if you look at the [download numbers](https://sourceforge.net/projects/manjarolinux/files/community/Cinnamon/2016.02/)!

**Comment by udeved** [comment]
I agree, hence I propose we release a 16.03 but only minimal profiles.
In my opinion, a good chance to see how well team works without Phil, annd it doesn't make sennse to me to roll back changes to a 15.12 outdated iso. :)

**Comment by oberon2007** [comment]
Well, but the full profiles need to be updated, too, no?

**Comment by udeved** [comment]
Yes, and no.
Assuming we do a 16.03, people will likely download 16.03.
Full profiles could be easily released with 16.06, and we'd have in the meantime updated install media we could refer to.

**Comment by oberon2007** [comment]
I don't agree. People will still download full profiles - most users want that, I think. Part of them will update in the live environment (which is a recommendation you often hear) They will install and the system will be broken. That's bad!

**Comment by oberon2007** [comment]
Building updated ISOs and uploading them as soon as the installers are in stable shouldn't be such a big deal! I think it should be done immediately.

**Comment by udeved** [comment]
What's the point of updating the live system? Its non persistent, will be lost if live session ends.

**Comment by oberon2007** [comment]
That's not the question. I don't do it. But people do! Watch some youtube-reviews! :laughing:

**Comment by udeved** [comment]
I don't concern myself with reviews, except if bugs and flaws are specifically pointed out. :smiling_imp:

**Comment by oberon2007** [comment]
Well the system will not boot. That's a bug :stuck_out_tongue_closed_eyes:

I only mentioned those videos because I've seen lots of them and I know that they often recommend to update the system before installing. - And maybe people just want to use the latest version of the installer for whatever reason. Calamares 2.0, hey!

**Comment by udeved** [comment]
Only if somebody does something stupid, such as updating live system, which is useless.
You can't prevent people from doing stupid things.

**Comment by udeved** [comment]
What I am saying, we are small team, the "chief" is away a couple of weeks, and I think its perfectly enough for us to target a minimal release only for the first time a release was done without Phil.
If we manage to get this done, perhaps ext time we can do it in full. I would consider it a test trial.

**Comment by oberon2007** [comment]
I know that you would hate this, but it is certainly an option to not update the installers now and roll back the tools to a state where the existing ISOs are update-safe and where we can still build installable ISOs now ... And then decide when is the right moment to transition to the new mhwd path.
I think the team should decide that carefully.

**Comment by udeved** [comment]
No way.
Kind of "back to the future".

**Comment by udeved** [comment]
I don't want to prevent anybody to do the full profiles too, but I won't.
I could do a 16.03 net,, perhaps kirek could take care for kde-minimal and the lsb-release stuff, and perhaps Berhard could do xfce.minimal.
If somebody volunteers for other profiles, fine with me. I just think we should not rush out too many iso when we don't have enough practice as team to coordinate releases yet.

**Comment by Ste74** [comment]
I like conservative way and  follow your discussion :wink: . I have express just my idea to split gnome in a minimal config...

**Comment by oberon2007** [comment]
To be honest: Aside from shorter building times I don't see such a big difference or advantage in building minimal instead of full profiles. In case of Cinnamon and Deepin I have just prepared the full profiles and then removed all the big packages ==> minimal. There's not really any difficult maintenance work with adding or removing gimp, steam, java &Co, is there? ... ;)

**Comment by oberon2007** [comment]
I mean to say. To prepare one or both in the end will not really make a huge difference.

**Comment by oberon2007** [comment]
I have offered to do Xfce already and I would not see a problem to prepare mini and maxi.

**Comment by udeved** [comment]
My point is less is more, more iso mean more support for each edition.
I view it as ideal test scenario to see how users react to a reduced set of editions.
Its just my opinion, that the huge iso are useless on a rolling distro.

**Comment by philmmanjaro** [comment]
Well, some changes are made and I didn't reviewed the source yet completely. Maybe I manage tomorrow. Today was hospital day. So I've to recover first a little ...

**Comment by udeved** [comment]
No worries Phil, everything is fine now, recover.
I sent two pull requests 5 minutes ago.

**Comment by philmmanjaro** [comment]
Ok. please build v0.9.4 of **thus** and **calamares 2.0-2** as needed. I just wondered why things didn't work as expected, that's all ...

**Comment by udeved** [comment]
Will do, but question is in the new pull requests.
Would be worth to decide on before rebuild.

**Comment by udeved** [comment]
Btw, I was also surprised by the little shocker, my bad, I never looked at the other installer before yesterday.

**Comment by philmmanjaro** [comment]
which pull requests?

**Comment by udeved** [comment]
2 new ones for thus and CAL.
I out a question/idea in the comment.

**Comment by philmmanjaro** [comment]
I'll do the installers now. Hopefully this will fix the current situation with **manjaro-tools 0.11.0** ...

**Comment by udeved** [comment]
I most probably will, the cli installer works, I had updated the package with new path.

**Comment by philmmanjaro** [comment]
... but simply forgot that we have also other installers. Some changes need to be discussed if other packages are also affected. I simply was wondering that happend and blamed **sddm** as I always test KDE for Blue Systems before pushing out some stable updates ...

**Comment by udeved** [comment]
I wasn't aware, thus and CAL were in fact depend on a static path on iso.
As said, my bad, I thought they download the driver.

**Comment by philmmanjaro** [comment]
All installers are in concept with offline mode. Therefore the path was fixed. Changes regarding **installer** need always to be discussed first. Else lot of people wonder what happened.

**Comment by philmmanjaro** [comment]
There are always options. Either fix all installers or create a symlink for backward compatibility as we are on a rolling release ...

**Comment by udeved** [comment]
Well, it times time to dive in code of 3 installers.
That happened yesterday.

**Comment by philmmanjaro** [comment]
That is why I meant to discuss those changes related to package overlays used by our installers. They are simply part of it. And if I'm not around @Kirek will be also glad to help you out with ...

**Comment by philmmanjaro** [comment]
@oberon2007: please check if [that fixes the matter](https://lists.manjaro.org/pipermail/manjaro-packages/Week-of-Mon-20160229/006182.html) ... If so, snap it over to **testing** compare it with **stable** and snap it over as a stable update incl. proper announcement in forums and blog. I've to take my meds now ...

**Comment by udeved** [comment]
I am saying, I simply did not have thus ad CAL o the radar, that they could be depend on a path set by manjaro-tools. This is noted since yesterday, and hence, it won't happen again adI propose we move the mwhd.shlater in manjaro-livecd-base.
Should we ever want to change this path, I would onnly impact the script base package the installers depend on.

**Comment by oberon2007** [comment]
ok for the update! will take care of it :)

**Comment by philmmanjaro** [comment]
Sure, we can do that @udeved. After the current updates in review are fixed. Sometimes it takes more time to find an issue made by one without discussing it with the team. Your changes are good and nobody is angry about it. However these things would have been prevented if we would have discussed it. Anyway. All is good. @oberon2007 knows how to push the update and I can relax, as I really need it right now. So I'll be off for some days cos my face is really swelling again and I've to take my meds and lay down ...

**Comment by udeved** [comment]
I said, I made a mistake, and the mistake was to not look earlier in the other two installers.
Then, these pull requested had been made when I changed the path in manjaro-tools.

**Comment by oberon2007** [comment]
Gute Besserung erst mal Philip! Wir werden den Kahn schon an die Hafenmauer steuern :grin:

**Comment by udeved** [comment]
Bernhard, please wait with the snapshot, I will eventually do a manjaro-tools-0.11.1 uin unstable.
I buiuld right now test iso with updated installers.
I fixed some minor stuff I found yesterday.

**Comment by udeved** [comment]
I can confirm calamares to be producing a booting system with sddm on systemd. Yay. :)

**Comment by oberon2007** [comment]
sure. I'm building a test ISO right now, too ... ;)

**Comment by udeved** [comment]
The can you test thus please?

**Comment by oberon2007** [comment]
Ah. cool. So are you still going to change any packages?

**Comment by udeved** [comment]
I will release ,amjaro-tools-0.11.1 in unstable, and you ca snap it over to testing, along with the installers. Thats what I meant.

**Comment by oberon2007** [comment]
Ok. I'll do that.

**Comment by oberon2007** [comment]
And I'll do that aswell, yes ;)

**Comment by udeved** [comment]
I was asking, if you can test thus on your test iso?
CAL works.

**Comment by oberon2007** [comment]
Yes, I will. That's what I meant, too! :laughing:

**Comment by udeved** [comment]
Overall, I think we are pretty good o target for 16.03.
Settings packages seem to work well now, iso profiles are small and stabilized.

I think we are doing good for a possible release next month.

**Comment by oberon2007** [comment]
Ok. You can uncork the Champagne now :stuck_out_tongue_winking_eye: 
**Thus** is now working as well :laughing: Congratulations, good job!
You can unleach the tools now, too!

**Comment by oberon2007** [comment]
I think it would be really easy to implement a security switch in the installers to keep them working for older install media, too: https://github.com/manjaro/thus/issues/45

**Comment by philmmanjaro** [comment]
@udeved and @oberon2007: please check if my current commits help you in some way: https://github.com/calamares/calamares-manjaro/commit/726ec3f81c10d24783c89710fc2a6bf85138058a https://github.com/manjaro/thus/commit/c513bbefdf3a9e82a19e1548b2a87dd77a473bbf

**Comment by udeved** [comment]
Looks good, now the script of CAL should be exactly the same as the one in thus.
It could be ported away.

**Comment by philmmanjaro** [comment]
Ok, please check and fast track these packages if needed: https://lists.manjaro.org/pipermail/manjaro-packages/Week-of-Mon-20160229/006190.html It's time for another shot of meds for me. I'm off again ...

**Comment by oberon2007** [comment]
Great, thanks! I'll verify with a 15.12 ISO with the updated Calamares first.

**Comment by philmmanjaro** [comment]
@oberon2007: you might need to remove **partitionmanager** first or update it also on a **15.12** release. Testing **thus** most likely is enough.

**Comment by oberon2007** [comment]
Well. Calamares apparently segfaults in the old environment. I'll have to do a full system update first to verify if that would work. I'll try Thus meanwhile...

**Comment by oberon2007** [comment]
ah. only saw your message now ;) Yes, thus is already busy installing a fluxbox 15.12 here ... ;)

**Comment by philmmanjaro** [comment]
Normally it is not meant to be updated on live-session. However only **thus** supports it.

**Comment by oberon2007** [comment]
Hm. Looks like we have a different problem in this scenario:
![flux1512](https://cloud.githubusercontent.com/assets/11800714/13424177/4018bf0e-df9f-11e5-8950-2b03968a1276.png)

**Comment by oberon2007** [comment]
It hangs there.

**Comment by oberon2007** [comment]
You can ignore the plymouth error of course. It always says that when you interrupt plymouth with a keystroke. But seems like something with **pam** or **lightdm** has changed since then?

**Comment by oberon2007** [comment]
Just booted into a console. Login with manjaro/manjaro is not accepted ...

**Comment by oberon2007** [comment]
I have no xfce 15.12 currently on my disk but have just asked Jonathon to try the same to rule out that is has something to do with just Fluxbox for whatever reason ...

**Comment by udeved** [comment]
xfce with lightdm works, also after install.

**Comment by oberon2007** [comment]
ok. so I'll forward the installers to testing and stable.

**Comment by jonathonf** [comment]
Installing Xfce 15.12 in VirtualBox (UK language, simple install, separate /home) results in a black screen with blinking cursor.

**Comment by oberon2007** [comment]
Oops! Thanks for double checking, @jonathonf ! What happens when you boot into console [like this](https://forum.manjaro.org/index.php?topic=31562.msg259403#msg259403) and then try to login?

**Comment by oberon2007** [comment]
@udeved are you sure you did install with an updated **thus**? :stuck_out_tongue_winking_eye:

**Comment by oberon2007** [comment]
If you're able to login like that @jonathonf , can you check if the video-drivers have been installed?

**Comment by jonathonf** [comment]
Console login works, which video drivers am I checking for?

**Comment by udeved** [comment]
No, I used calamares.

**Comment by udeved** [comment]
Did you guys both build with enabled plymouth?

**Comment by jonathonf** [comment]
I installed with the release 15.12 iso and an updated Thus.

**Comment by udeved** [comment]
I see. 
It gets confusing to have more issues in one ticket.

**Comment by jonathonf** [comment]
Ah, then I'm making noise due to @oberon2007 . :)

**Comment by udeved** [comment]
No problem, but we should split up the issues and open tickets for each.
I just fixed a plymouth thingy, silly me forget to call the function properly..

**Comment by oberon2007** [comment]
why different issues? we are trying to verify if the fix in the installers works! its about old install media with updated installers: https://github.com/manjaro/release-plan/issues/33#issuecomment-190625163

**Comment by udeved** [comment]
Well, then please open issues if you report eg lightdm issues.
Its really hard to keep track this way, if we have multiple issues here.

**Comment by oberon2007** [comment]
we are not building anything, @udeved ! We are trying to install 15.12 ISOs with **thus** updated in the live environment. What are you talking about?

**Comment by Kirek** [comment]
Don't change the topic if the issue then.

**Comment by oberon2007** [comment]
i am not reporting lightdm issues. i am reporting the result of my test.

**Comment by udeved** [comment]
You do follow what you post here?

I quote:

>You can ignore the plymouth error of course. It always says that when you interrupt plymouth with a keystroke. But seems like something with pam or lightdm has changed since then?

**Comment by oberon2007** [comment]
the important information however is: **the "fixed" installers appear to not be able to install video drivers in 15.12 install media.**

**Comment by udeved** [comment]
More important than fixing the install media so they boot up the plymouth-dm service properly?

**Comment by oberon2007** [comment]
yes. ignore that. it was just an explanation.
other than that, do with the information what you like.

**Comment by udeved** [comment]
Well, its good I didn't ignore it but fixed it. You tell me, if I should ignore issues in the future.

**Comment by Kirek** [comment]
Maybe we should just create an new issue for plymouth if its broken...

**Comment by oberon2007** [comment]
plymouth is working properly! it is supposed to stop when you hit a key. but this is not the topic. topic is: latest **thus** is not able to install video-drivers on 15.12 ISOs

**Comment by udeved** [comment]
You seem to have moire code insights than me.
Its did not work fully inn certain scenarios, only if you had plymouth installed on your system.
Fixed now.

**Comment by oberon2007** [comment]
the fix in thus appears to not be working!

**Comment by udeved** [comment]
Whatever, one issue fixed, one to go.

**Comment by udeved** [comment]
So I am confused now.
Where is the this build that adds a symlink?

either the package i the repo only has my additions ad no symlink, or the PKGBUILD is not updated in git.

**Comment by Kirek** [comment]
https://github.com/manjaro/thus/commit/c513bbefdf3a9e82a19e1548b2a87dd77a473bbf

**Comment by udeved** [comment]
Well, I know why it doesn't work.

in  thus/installation/process.py 

line 1020

**Comment by oberon2007** [comment]
good. :wink:

**Comment by udeved** [comment]
Yes. As mentioned, I looked at the code now. :grimacing:

**Comment by udeved** [comment]
https://github.com/manjaro/thus/commit/1e5957362933b678eeda02b5d46de307146b7995

**Comment by Kirek** [comment]
@oberon2007 @jonathonf 
if you have time can you check if is solved with thus 0.9.4.2?

**Comment by philmmanjaro** [comment]
I also updated **calamares** with given changes. However **calamares** won't work at all with older install medias, as it is compiled to the given **kde framework** ...

**Comment by jonathonf** [comment]
15.12 Xfce with Thus 0.9.4.2 installs and boots correctly (VirtualBox, simple install). Looks good to me.

**Comment by Kirek** [comment]
Thanks for testing, I think this is solved.

**Comment by oberon2007** [comment]
Yes, thank you all!

**Comment by udeved** [comment]
I have added old path support also to manjaro-live-base package.

If you have time, please test the new packages by changig Packages-Livecd into:

~~~
>openrc manjaro-live-openrc
>systemd manjaro-live-systemd
manjaro-cli-installer
manjaro-live-skel		# live skel
calamares			# new GUI installer of manjaro
#calamares-dev			# new GUI installer of manjaro - kpmcore
#calamares-exp			# new GUI installer of manjaro - experimental version
thus
#thus-dev
#gparted
~~~

If everything works, we can remove the manjaro-livecd-* packages from repo.

**Comment by udeved** [comment]
The livecd repo would be in shape now to host a common installer script, that sets the repo path for all 3 installers.
We basically have pretty similar scripted tasks in all 3 installers. 
We could also use the new live packages to test this in parallel, and delete "old" livecd packages later when its all working.
